core/src/main/java/org/bitcoinj/core/NetworkParameters.java

```diff
@@ -158,26 +161,30 @@ public String getSporkAddress() {
         return strSporkAddress;
     }
 
+    public int getMinSporkKeys() {
+        return minSporkKeys;
+    }
+
     protected NetworkParameters() {
         alertSigningKey = SATOSHI_KEY;
         genesisBlock = createGenesis(this);
     }
-    //TODO:  put these bytes into the CoinDefinition
+
     private static Block createGenesis(NetworkParameters n) {
         Block genesisBlock = new Block(n, Block.BLOCK_VERSION_GENESIS);
         Transaction t = new Transaction(n);
         try {
             // A script containing the difficulty bits and the following message:
             //
             //   coin dependent
-            byte[] bytes = Utils.HEX.decode(CoinDefinition.genesisTxInBytes);
+            byte[] bytes = Utils.HEX.decode("04ffff001d01044c5957697265642030392f4a616e2f3230313420546865204772616e64204578706572696d656e7420476f6573204c6976653a204f76657273746f636b2e636f6d204973204e6f7720416363657074696e6720426974636f696e73");
 
             t.addInput(new TransactionInput(n, t, bytes));
             ByteArrayOutputStream scriptPubKeyBytes = new ByteArrayOutputStream();
-            Script.writeBytes(scriptPubKeyBytes, Utils.HEX.decode(CoinDefinition.genesisTxOutBytes));
+            Script.writeBytes(scriptPubKeyBytes, Utils.HEX.decode("040184710fa689ad5023690c80f3a49c8f13f8d45b8c857fbcbc8bc4a8e4d3eb4b10f4d4604fa08dce601aaf0f470216fe1b51850b4acf21b179c45070ac7b03a9"));
```

Comment:
Are we sure we want to use magic strings here instead of old consts? At least the naming made what's happening a bit more clear. If we don't need `CoinDefinition` anymore, maybe just create some `const` values here or in a new class.

Reply:
Some history: I created `CoinDefinition` about 7 years ago to make it easier to make forks of `bitcoinj` for other coins.  All of the values that make a coin unique would be in one class.  The values within this class are considered as constants since they define parameters that make Dash unique from Bitcoin.  

I thought to remove this class since it was no longer updated and its existence became obsolete.

What I can do is for some of these is make them into `const` values as well as add comments explaining what they are for or where they come from.  This will be done in the next phase of upgrades for Dash Core 0.18.  A note was marked in the story.

The one above `CoinDefinition.genesisTxOutBytes` is only used in one place and is mirrored here in the Dash Core Reference code:  https://github.com/dashpay/dash/blob/master/src/chainparams.cpp#L65-L81

Many of the `CoinDefinition` class fields came from this same C++ source file.

Commit: https://github.com/dashevo/dashj/pull/138/commits/72b3aadfaef43c603d701c0ed840b081ac361115